extensions/wikia/TemplateDraft/ApprovedraftAction.php

```diff
@@ -70,12 +70,15 @@ private function approveDraft( Title $draftTitle ) {
 		global $wgEnableInsightsInfoboxes;
 
 		// Get Title object of parent page
-		$helper = new TemplateDraftHelper();
-		$parentTitle = $helper->getParentTitle( $draftTitle );
+		$parentTitle = TemplateDraftHelper::getParentTitle( $draftTitle );
 
 		// Check edit rights
-		if ( !$parentTitle->userCan( 'templatedraft' ) ) {
-			throw new PermissionsException( 'edit' );
+		if ( !$parentTitle->userCan( 'templatedraft' ) || !$parentTitle->userCan( 'edit' ) ) {
+			throw new ErrorPageError( 'badaccess', 'templatedraft-protect-edit' );
+		}
+
+		if ( !$parentTitle->userCan( 'move' ) ) {
```

Comment:
And actually, this would likely break functionality if a template was `move` protected, but not `edit` protected (which is somewhat common), as then a user wouldn't be able to approve a draft for template they can edit, just because they can't move it.